src/nccl_ofi_rdma.c

```diff
@@ -5403,6 +5392,32 @@ static inline int set_local_address(struct fid_ep *ep, nccl_net_ofi_ep_rail_t *r
 	return 0;
 }
 
+
+static int ep_rail_init(nccl_net_ofi_rdma_ep_t *ep,
+			int dev_id, int rail_id,
+			nccl_net_ofi_rdma_device_rail_t *dev_rail,
+			nccl_net_ofi_ep_rail_t *ep_rail)
+{
+	int ret = 0;
+
+	ret = nccl_ofi_ofiutils_init_connection(FI_VERSION(1, 18),
+						dev_rail->info, dev_rail->domain,
+						&ep_rail->ofi_ep, &ep_rail->av, &ep_rail->cq);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ep_rail->rail_id = rail_id;
+
+	ret = set_local_address(ep_rail->ofi_ep, ep_rail);
+	if (ret != 0) {
+		ep_rail_release(ep_rail, dev_id);
+	}
+
+	return 0;
```

Comment:
don't we need to return an error if `set_local_address` fails?

Reply:
yup, missed that.  fixed.